Guide/validation.markdown

```diff
@@ -250,6 +250,116 @@ user
 
 In this example, when the [`validateIsUnique`](https://ihp.digitallyinduced.com/api-docs/IHP-ValidationSupport-ValidateIsUnique.html#v:validateIsUnique) function adds an error to the user, the message `Email Has Already Been Used` will be used instead of the default `This is already in use`.
 
+## Security Concerns and Conditional `fill`
+
+It's important to remember that any kind of validations you might have on the form level are not enough to ensure the security of your application. You should always have validations on the backend as well. The user might manipulate the form data and send invalid data to your application.
+
+So if a field is disabled or has an integer field with a min/max value, you should always have a validation on the backend.
+
+Another point is that you don't have to always `fill` all fields in one go. Sometimes you'd like to conditionally `fill` based on the current user or based on the current logic.
+
+Let's see those examples in action. Let's say we have a `Comment` record that has a `postId` that references a `Post`, a `body` field, and a moderation field allowing admin users to indicate if they are approved or rejected.
+
+Here's an excerpt from the `Schema.sql`:
+
+```sql
+-- Schema.sql
+
+-- Add a moderation status column to the comments table
+CREATE TYPE comment_moderation AS ENUM ('comment_moderation_pending', 'comment_moderation_approved', 'comment_moderation_rejected');
+
+CREATE TABLE comments (
+    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY NOT NULL,
+    post_id UUID NOT NULL,
+    body TEXT NOT NULL,
+    created_at TIMESTAMP WITH TIME ZONE DEFAULT NOW() NOT NULL
+    comment_moderation comment_moderation NOT NULL
+);
+```
+
+Once we generate a controller for the `Comment` record, we'll the `buildComment` function that is called from both the `CreateCommentAction` and the `UpdateCommentAction`:
+
+```haskell
+-- Controller/Comments.hs
+
+buildComment comment = comment
+    |> fill @'["postId", "body", "commentModeration"]
+```
+
+We'll start with the `postId`. Once a comment is refernecing a post it will never have the reference change. So it means we should fill it only upon creation.
+
+```haskell
+buildComment comment = comment
+    |> fill @'["body", "commentModeration"]
+    |> fillIsNew
+    where
+        fillIsNew record
+            -- Record is new, so fill the `postId`.
+            | isNew record = fill @'["postId"] record
+            -- Otherwise, leave the record as is.
+            | otherwise = record
+```
```

Comment:
Likely it would be better if we stick to normal `if .. then .. else ..` in the documentation to avoid more complex syntax beginners might not be aware about

Reply:
Done